File: src/main/java/com/travel/domain/city/controller/CityController.java
package com.travel.domain.city.controller;

import com.travel.domain.city.dto.CitySaveDto;
import com.travel.domain.city.dto.CityDto;
import com.travel.domain.city.dto.CityUpdateDto;
import com.travel.domain.city.service.CityService;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.*;

@RequiredArgsConstructor
@RestController
public class CityController {
    private final CityService cityService;

    @PostMapping("/city/save")
    public Long save(@RequestBody CitySaveDto requestDto) {
        return cityService.save(requestDto);
    }

    @PutMapping("/city/update/{id}")
    public void update(@PathVariable Long id, @RequestBody CityUpdateDto cityUpdateDto) {
        cityService.update(id, cityUpdateDto);
    }

    @DeleteMapping("/city/delete/{id}")
    public void delete(@PathVariable Long id) {
        cityService.delete(id);
    }

    @PostMapping("/city/get/{id}")
    public CityDto get(@PathVariable Long id) {
        return cityService.get(id);
    }
}

File: src/main/kotlin/com/travel/domain/city/controller/CityController.kt
package com.travel.domain.city.controller

import com.travel.domain.city.dto.CityDto
import com.travel.domain.city.dto.CitySaveDto
import com.travel.domain.city.dto.CityUpdateDto
import com.travel.domain.city.service.CityService
import com.travel.global.base.BaseResponse
import org.springframework.web.bind.annotation.*

@RestController
class CityController(
        private val cityService: CityService
) {
    @PostMapping("/city/save")
    fun save(@RequestBody requestDto: CitySaveDto): BaseResponse<Unit> {
        cityService.save(requestDto)
        return BaseResponse(message = "도시 ${requestDto.name}의 등록이 완료되었습니다.")
    }

    @PutMapping("/city/update/{id}")
    fun update(@PathVariable id: Long, @RequestBody cityUpdateDto: CityUpdateDto): BaseResponse<Unit> {
        cityService.update(id, cityUpdateDto)
        return BaseResponse(message = "도시가 수정되었습니다.")
    }

    @DeleteMapping("/city/delete/{id}")
    fun delete(@PathVariable id: Long): BaseResponse<Unit> {
        cityService.delete(id)
        return BaseResponse(message = "도시가 삭제되었습니다.")
    }

    @PostMapping("/city/get/{id}")
    fun get(@PathVariable id: Long): BaseResponse<CityDto> {
        return BaseResponse(data = cityService.get(id))
    }

    @PostMapping("/city/findCityListByUserIdNative")
    fun findCityListByUserIdNative(@PathVariable userId: String): BaseResponse<List<Map<String, Any>>> {
        return BaseResponse(data = cityService.findCityListByUserIdNative(userId))
    }
}